sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs

```diff
@@ -1348,13 +1350,6 @@ private void OnReceiverLinkClosed(object receiver, EventArgs e)
                 LinkException = exception;
             }
 
-            if (IsSessionLinkClosed)
```

Comment:
Even if the underlying session link closed, the user still needs to dispose the receiver.

Reply:
Reverted this change as it will be too confusing for users.